services/catarse/db/migrate/20200303163420_insert_user_current_bank_account_on_balance_transfer.rb

```diff
@@ -0,0 +1,123 @@
+class InsertUserCurrentBankAccountOnBalanceTransfer < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+
+CREATE OR REPLACE FUNCTION public.withdraw_balance()
+    RETURNS trigger
+    LANGUAGE plpgsql
+AS $function$
+    DECLARE
+        v_balance "1".balances; 
+        v_balance_transfer_id integer; 
+        v_balance_transfer "1".balance_transfers; 
+        v_balance_transaction public.balance_transactions;
+        bank_json jsonb;
+    BEGIN 
+        IF NOT public.is_owner_or_admin(NEW.user_id) THEN 
+            RAISE EXCEPTION 'insufficient privileges to insert balance_transactions'; 
+        END IF; 
+        SELECT * FROM "1".balances 
+            WHERE user_id = NEW.user_id 
+            INTO v_balance; 
+        IF COALESCE(v_balance.amount, 0) <= 0 THEN 
+            RAISE EXCEPTION 'insufficient funds'; 
+        END IF; 
+        IF v_balance.in_period_yet is not null AND v_balance.in_period_yet THEN
+          RAISE EXCEPTION 'already requested transfer this month'; 
+        END IF;
+        
+      SELECT json_build_object(
+          'agency', ba.agency,
+          'agency_digit', ba.agency_digit,
+          'account', ba.account,
+          'account_digit', ba.account_digit,
+          'bank_account_type', ba.account_type,
+          'document_type', CASE WHEN u.account_type = 'pf' THEN 'cpf' ELSE 'cnpj' END,
+          'document_number', regexp_replace(u.cpf, '[^0-9]+', '', 'g'),
+          'name', u.name
+      ) FROM public.bank_accounts ba INNER JOIN users u ON ba.user_id = NEW.user_id AND u.id = NEW.user_id LIMIT 1
+      INTO bank_json;
+        
+        INSERT INTO public.balance_transfers (user_id, amount, created_at, metadata) 
```

Comment:
When creating a new balance_transfers row, added the new column "metadata" with bank_json pre-filled.